### PR TITLE
Move Documentation Job to GHA

### PR DESCRIPTION
**Pull request by assignUser**
This PR moves the doc gen job to gha. There are some minor changes compared to the original job:
- it uses the built-in `GITHUB_TOKEN` so it will no longer be necessary to keep an SSH key with full write access around.
  - The key currently used by CCI should be deleted once this is merged. 
- only runs when changes happen in velox/docs
- runs in relevant PRs and uploads the generated docs as a shortlived artifact for easier review
- I removed the pyvelox build for now as the the API documentation wasn't generated/used. We can add this in when we want to use it, until then it's just wasted CI time.

I have also added a `dependabot.yml` to enable dependabot for Github Actions updates (this might need to be enabled in the repo settings). To avoid supply chain attacks via force pushed tag of actions we want to pin actions to a specific SHA when using actions in privileged workflows with permissions (e.g. `contents: write`) or that expose secrets. Dependabot will open PRs to update these SHAs (and the normal version tags)

### CODE REVIEW

**Comment by kgpai** [comment]
Hi @assignUser , Is there anyway to test this before merging ?

**Comment by assignUser** [comment]
You can see the deployment working in my fork: https://github.com/assignUser/velox/actions/runs/7851419513/job/21428469902 with the result: https://assignuser.github.io/velox/
But the job also uploads the build docs as an artifact when run in PRs: https://github.com/facebookincubator/velox/actions/runs/7851632833?pr=8718

**Review by kgpai — APPROVED** [review]

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+version: 2
+updates:
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "weekly"
```
Comment (kgpai) [inline .github/dependabot.yml:20]: nit: what is this job for ?

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  cancel-in-progress: true
+
+jobs:
+  build_docs:
+    name: Build and Push
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```
Comment (kgpai) [inline .github/workflows/docs.yml:42]: nit: why do we need to checkout such specific versions of 'checkout' ?

File: .github/dependabot.yml (hunk 1 above)
Comment (assignUser) [reply .github/dependabot.yml:20]: See my other comment.

**Pull request closed without merging**